update_copyright.d

```diff
@@ -0,0 +1,358 @@
+    static string[] skipDirs = [
+        "docs",
+        "ini",
+        "test",
+        "samples",
+        "vcbuild",
+    ];
+
+    static string[] skipFiles = [
+        "Jenkinsfile",
+        "LICENSE.txt",
+        "VERSION",
+        ".a",
+        ".ddoc",
+        ".deps",
+        ".lst",
+        ".map",
+        ".md",
+        ".o",
+        ".obj",
+        ".sdl",
+        ".sh",
+        ".yml",
+    ];
+
+    // Characters in a range of years.
+    // Include '.' for typos, and '?' for unknown years.
+    enum rangesStr = `[0-9?](?:[-0-9.,\s]|\s+and\s+)*[0-9]`;
+
+    // Non-whitespace characters in a copyright holder's name.
+    enum nameStr = `[\w.,-]`;
+
+    // Matches a full copyright notice:
+    // - 'Copyright (C)', etc.
+    // - The years. Includes the whitespace in the year, so that we can
+    //   remove any excess.
+    // - 'by ', if used
+    // - The copyright holder.
+    static copyrightRe = ctRegex!(`([Cc]opyright` ~ `|[Cc]opyright\s+\([Cc]\))` ~
+                                  `(\s*(?:` ~ rangesStr ~ `,?)\s*)` ~
+                                  `(by\s+)?` ~
+                                  `(` ~ nameStr ~ `(?:\s?` ~ nameStr ~ `)*)?`);
+
+    // A regexp for notices that might have slipped by.
+    static otherCopyrightRe = ctRegex!(`copyright.*[0-9][0-9]`, `i`);
+
+    // A regexp that matches one year.
+    static yearRe = ctRegex!(`[0-9?]+`);
+
+    // Matches part of a year or copyright holder.
+    static continuationRe = ctRegex!(rangesStr ~ `|` ~ nameStr);
+
+    static commentRe = ctRegex!(`#+|[*]+|;+|//+`);
+
+    // Convenience for passing around file/line number information.
+    struct FileLocation
+    {
+        string filename;
+        size_t linnum;
+
+        string toString()
+        {
+            import std.format;
+            return "%s(%d)".format(this.filename, this.linnum);
+        }
+    }
+
+    void processFile(string file)
+    {
+        // Looks like something we tried to create before.
+        if (file.endsWith(".tmp"))
+        {
+            remove(file);
+            return;
+        }
+
+        auto h = File(file, "rb");
+        auto lines = appender!string;
+        int errors = 0;
+        bool changed = false;
+
+        lines.reserve(h.size);
+
+        char[] orig_line = null;
+        char[] next_line = null;
+
+        auto loc = FileLocation(file, 0);
+
+        foreach (line; h.byLine)
+        {
+            loc.linnum++;
+            try
+            {
+                if (orig_line)
+                {
+                    auto continuation = this.stripContinuation(line);
+                    if (!continuation.matchAll(continuationRe))
+                    {
+                        // If the next line doesn't look like a proper continuation,
+                        // assume that what we've got is complete.
+                        lines.put(orig_line);
+                        lines.put('\n');
```

Comment:
```d
import std.ascii : newline;
lines.put(newline);
```

Reply:
I disagree with this.  We use `\n` in all dmd sources.  Making it `std.ascii.newline` would mean that someone processing all sources on Windows would have a different output than someone on Linux.